components/prism-core.js

```diff
@@ -395,6 +675,15 @@ var _ = {
 			hooks[name].push(callback);
 		},
 
+		/**
+		 * Runs a hook invoking all registered callbacks with the given environment variables.
+		 *
+		 * Callbacks will be invoked synchronously and in the order in which they were registered.
+		 *
+		 * @param {string} name The name of the hook.
+		 * @param {Object<string, any>} env The environment variables of the hook passed to all callbacks registered.
```

Comment:
Can we define the `env` structure more explicitly?

Reply:
No. You can do custom hooks (see Line numbers), so we technically can't even assume that `env` is an object.